ЛР_10

IvanovDmitr opened the pull request.

IvanovDmitr committed:
ЛР_10

IvanovDmitr committed:
Add files via upload

NemirovichMark commented:
Здравствуйте, Дмитрий.
Сортировки написать самостоятельные
 public void Sort(string criteria)
 {
     if (criteria == "points")
     {
         Teams.Sort((x, y) => y.CalculatePoints().CompareTo(x.CalculatePoints()));
     }
     else if (criteria == "name")
     {
         Teams.Sort((x, y) => x.Name.CompareTo(y.Name));
     }
 }

 public void Sort(int dummy)
 {
     if (dummy == 0)
     {
         Teams.Sort((x, y) => y.CalculatePoints().CompareTo(x.CalculatePoints()));
     }
     else
     {
         Teams.Sort((x, y) => x.Name.CompareTo(y.Name));
     }
 }

Не увидел файла для:
Создайте 2-й partial класс TournamentTable. В нем создайте перегруженные методы GetBest(), GetBest(int), которые выводят лучшую команду или лучшие n команд.

Составьте UML-диаграммы **классов, процессов и состояний.** Не нашел у вас.

На доработку.

IvanovDmitr committed:
Add files via upload

Мы сделали все диаграммы кроме состояний. Нужно отправлять те две, если одной не достаёт?

IvanovDmitr committed:
Add files via upload

Диграммы

NemirovichMark commented:
Здравствуйте, Дмитрий.
Так, теперь тут использовать вашу написанную сортировку
 public Team GetBest()
        {
            Teams.Sort((x, y) => y.CalculatePoints().CompareTo(x.CalculatePoints()));
            return Teams[0];
        }

        public List<Team> GetBest(int n)
        {
            Teams.Sort((x, y) => y.CalculatePoints().CompareTo(x.CalculatePoints()));
            return Teams.GetRange(0, Math.Min(n, Teams.Count));
        }

Добавить проверку, что такая команда есть в списке, чтобы её можно было удалять
  public void Disqual(Team team)
        {
            Teams.Remove(team);
        }

Волейбол не наследуется от баскетбола. Интерфейс не связан с командой, связан класс таблицы. Причем нужно указать правильный тип связи
![image](https://github.com/NemirovichMark/BBI-23-6/assets/112826355/3bebff8e-4748-4972-b43c-51bef67e0834)

Добавить main. И уже им связать все нужные классы (экземпляры которых в нем создаются)
![image](https://github.com/NemirovichMark/BBI-23-6/assets/112826355/6ffe6dfc-32a7-4525-ac14-8a53fbe96e6d)

В диаграмме процессов потеряна связь между объектами. Не нужно такие полосочки делать и связывать. Прям блоки с действиями должны быть и переходы от одного блока (метода) к другому (в нужном классе)
![image](https://github.com/NemirovichMark/BBI-23-6/assets/112826355/7a8c878c-036f-4d54-a78a-cda6007b4576)

Диаграмму состояний не увидел.

На доработку.
